### ISSUE TITLE
proxy_pass UI does not work

### ISSUE THREAD

**Issue by Alexey-Tsarev** [WeeJeWel/wg-easy#223]
Hi, I'd like to hide UI behind Nginx with SSL enabled, because the UI works with the plain HTTP - it's not good!

It works, but I want to use some prefix (to hide UI). Let me demo (I am using a self-singing cert for this demo and the `/wG` prefix):
<img width="866" alt="image" src="https://user-images.githubusercontent.com/5929124/158019070-afa24dce-5d32-45fb-92ca-e48e717b50f8.png">

```
[12/Mar/2022:17:05:00 +0400] 10.1.1.25 TLSv1.2 10.1.1.101 - "GET /wG/ HTTP/2.0" 200 6334 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/98.0.4758.102 Safari/537.36" "-"
[12/Mar/2022:17:05:00 +0400] 10.1.1.25 TLSv1.2 10.1.1.101 - "GET /wG/img/logo.png HTTP/2.0" 200 3103 "https://10.1.1.101/wG/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/98.0.4758.102 Safari/537.36" "-"
[12/Mar/2022:17:05:00 +0400] 10.1.1.25 TLSv1.2 10.1.1.101 - "GET /wG/js/vendor/vue.min.js HTTP/2.0" 200 93670 "https://10.1.1.101/wG/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/98.0.4758.102 Safari/537.36" "-"
[12/Mar/2022:17:05:00 +0400] 10.1.1.25 TLSv1.2 10.1.1.101 - "GET /wG/js/vendor/vue-apexcharts.min.js HTTP/2.0" 200 4464 "https://10.1.1.101/wG/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/98.0.4758.102 Safari/537.36" "-"
[12/Mar/2022:17:05:00 +0400] 10.1.1.25 TLSv1.2 10.1.1.101 - "GET /wG/js/vendor/md5.min.js HTTP/2.0" 200 3734 "https://10.1.1.101/wG/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/98.0.4758.102 Safari/537.36" "-"
[12/Mar/2022:17:05:00 +0400] 10.1.1.25 TLSv1.2 10.1.1.101 - "GET /wG/css/vendor/tailwind.min.css HTTP/2.0" 200 2927531 "https://10.1.1.101/wG/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/98.0.4758.102 Safari/537.36" "-"
[12/Mar/2022:17:05:00 +0400] 10.1.1.25 TLSv1.2 10.1.1.101 - "GET /wG/js/vendor/apexcharts.min.js HTTP/2.0" 200 485189 "https://10.1.1.101/wG/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/98.0.4758.102 Safari/537.36" "-"
[12/Mar/2022:17:05:00 +0400] 10.1.1.25 TLSv1.2 10.1.1.101 - "GET /wG/js/vendor/timeago.min.js HTTP/2.0" 200 1740 "https://10.1.1.101/wG/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/98.0.4758.102 Safari/537.36" "-"
[12/Mar/2022:17:05:00 +0400] 10.1.1.25 TLSv1.2 10.1.1.101 - "GET /wG/js/api.js HTTP/2.0" 200 2345 "https://10.1.1.101/wG/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/98.0.4758.102 Safari/537.36" "-"
[12/Mar/2022:17:05:00 +0400] 10.1.1.25 TLSv1.2 10.1.1.101 - "GET /wG/js/app.js HTTP/2.0" 200 8902 "https://10.1.1.101/wG/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/98.0.4758.102 Safari/537.36" "-"
[12/Mar/2022:17:05:01 +0400] 10.1.1.25 TLSv1.2 10.1.1.101 - "GET /api/session HTTP/2.0" 404 351 "https://10.1.1.101/wG/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/98.0.4758.102 Safari/537.36" "-"
[12/Mar/2022:17:05:01 +0400] 10.1.1.25 TLSv1.2 10.1.1.101 - "GET /api/release HTTP/2.0" 404 350 "https://10.1.1.101/wG/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/98.0.4758.102 Safari/537.36" "-"
```

Last two lines contain `/api/session`, `/api/release`. Also there is the following endpoint `/api/wireguard/client`
**As you may see, the UI forms links incorrectly**: it should call `/wG/api/session` and `/wG/api/release`

It's better to have a UI version which supports a relative links. In this case, it's easy to hide UI behind reverse proxy.

I have done with the following config:
```
server {
    listen 80  default_server reuseport;
    listen 443 default_server reuseport ssl http2;

    server_name default.biz www.default.biz;

    root /home/www_default/public_html/www;

    location /wG/ {
        proxy_pass http://127.0.0.1:51821/;

        proxy_set_header Host              $host;
        proxy_set_header X-Forwarded-Proto $scheme;
        proxy_set_header X-Forwarded-Host  $http_host;
        proxy_set_header X-Forwarded-For   $http_host;
        proxy_set_header X-Real-IP         $http_host;
    }

    location /api {
        proxy_pass http://127.0.0.1:51821/api;

        proxy_set_header Host              $host;
        proxy_set_header X-Forwarded-Proto $scheme;
        proxy_set_header X-Forwarded-Host  $http_host;
        proxy_set_header X-Forwarded-For   $http_host;
        proxy_set_header X-Real-IP         $http_host;
    }
```

Result:
<img width="733" alt="image" src="https://user-images.githubusercontent.com/5929124/158019393-0bf636e2-0136-4902-8b61-a70fffb27a8e.png">

And the goal is to hide UI and place in site root something "harmless" (for example broken WordPress install):
<img width="475" alt="image" src="https://user-images.githubusercontent.com/5929124/158019479-3be95531-875b-4845-8334-308802a73a8f.png">

But it's better to not have the config:
```
location /api
```
but currently it's impossible due to UI constructs links incorrectly.

**Comment by nemomode7** [comment]
What about using subdomains instead? Something like `status.example.com`? You also can hide the service behind basic auth, so noone finds out what's it is actually for

**Comment by thepheer** [comment]
Here's my simple referer-based workaround for the time being:

```nginx
if ($http_referer ~ /wg/) {
  rewrite ^(?!/wg)(.*)$ /wg$1;
}

location /wg/ {
  proxy_pass http://wg-easy:51821/;
}
```

**Comment by WeeJeWel** [comment]
Fixed in e7f4cc2d56fe83e629166795db432ea203e635fd
